geom/type_point.go

```diff
@@ -19,18 +19,9 @@ type Point struct {
 }
 
 // NewPoint creates a new point gives its Coordinates.
-func NewPoint(c Coordinates, _ ...ConstructorOption) Point {
-	return Point{c, true}
-}
-
-// NewEmptyPoint creates a Point that is empty.
-func NewEmptyPoint(ctype CoordinatesType) Point {
-	return Point{Coordinates{Type: ctype}, false}
-}
-
-// NewPointFromXY creates a new point from an XY.
-func NewPointFromXY(xy XY, _ ...ConstructorOption) Point {
-	return Point{Coordinates{XY: xy, Type: DimXY}, true}
+func NewPoint(oc OptionalCoordinates, _ ...ConstructorOption) Point {
+	c := Coordinates{oc.XY, oc.Z, oc.M, oc.Type}
+	return Point{c, !oc.Empty}
```

Comment:
Something to point out here is that the `bool` used to indicate empty in `Point` and in `OptionalCoordinates` has flipped meaning. In `Point`, the bool is `full` and in `OptionalCoordinates` the bool is `Empty`.

There are a few reasons for this:

- It's important for the zero value of `Point` to represent `POINT EMPTY`. The zero value for all other concrete geometry types is the empty XY value of that geometry. So therefore `false` must encode emptiness.
- When specifying an `OptionalCoordinates`, it's intended that users can specify _either_ `Empty: true` _or_ `XY: geom.XY{X:2, Y:3}` (and then also `Z` and `M` if appropriate). This makes things a bit more ergonomic from the perspective of users. If the `bool` had the same meaning as in `Point`, then the user would have to specify `Populated: true, XY: geom.XY{...}` for populated points, and no fields at all for empty points. It could be fairly easy for users to miss the `Populated: true` part, and _think_ they are creating a populated point (but really it would be empty).